test/gtest/core/test_mc_reduce.cc

```diff
@@ -0,0 +1,257 @@
+                                           ReductionTest<UCC_DT_INT32, min>,
+                                           ReductionTest<UCC_DT_INT64, min>,
+                                           ReductionTest<UCC_DT_INT16, sum>,
+                                           ReductionTest<UCC_DT_INT32, sum>,
+                                           ReductionTest<UCC_DT_INT64, sum>,
+                                           ReductionTest<UCC_DT_INT16, prod>,
+                                           ReductionTest<UCC_DT_INT32, prod>,
+                                           ReductionTest<UCC_DT_INT64, prod>,
+                                           ReductionTest<UCC_DT_INT16, land>,
+                                           ReductionTest<UCC_DT_INT32, land>,
+                                           ReductionTest<UCC_DT_INT64, land>,
+                                           ReductionTest<UCC_DT_INT16, band>,
+                                           ReductionTest<UCC_DT_INT32, band>,
+                                           ReductionTest<UCC_DT_INT64, band>,
+                                           ReductionTest<UCC_DT_INT16, lor>,
+                                           ReductionTest<UCC_DT_INT32, lor>,
+                                           ReductionTest<UCC_DT_INT64, lor>,
+                                           ReductionTest<UCC_DT_INT16, bor>,
+                                           ReductionTest<UCC_DT_INT32, bor>,
+                                           ReductionTest<UCC_DT_INT64, bor>,
+                                           ReductionTest<UCC_DT_INT16, lxor>,
+                                           ReductionTest<UCC_DT_INT32, lxor>,
+                                           ReductionTest<UCC_DT_INT64, lxor>,
+                                           ReductionTest<UCC_DT_INT16, bxor>,
+                                           ReductionTest<UCC_DT_INT32, bxor>,
+                                           ReductionTest<UCC_DT_INT64, bxor>,
+                                           ReductionTest<UCC_DT_FLOAT32, max>,
+                                           ReductionTest<UCC_DT_FLOAT64, max>,
+                                           ReductionTest<UCC_DT_FLOAT32, min>,
+                                           ReductionTest<UCC_DT_FLOAT64, min>,
+                                           ReductionTest<UCC_DT_FLOAT32, sum>,
+                                           ReductionTest<UCC_DT_FLOAT64, sum>,
+                                           ReductionTest<UCC_DT_FLOAT32, prod>,
+                                           ReductionTest<UCC_DT_FLOAT64, prod>>;
+
+TYPED_TEST_CASE(test_mc_reduce, ReductionTypesOps);
+TYPED_TEST(test_mc_reduce, ucc_op_sum) {
+    ucc_mc_reduce(this->buf1, this->buf2, this->res,
+                  this->COUNT, TypeParam::dt,
+                  TypeParam::redop, UCC_MEMORY_TYPE_HOST);
```

Comment:
what about MEMORY_TYPE_CUDA?

Reply:
What about reduce_multi?

Reply:
Yes, if overall approach looks good I will add cuda and reduce multi